src/osx/cocoa/textctrl.mm

```diff
@@ -988,10 +988,10 @@ - (void)controlTextDidEndEditing:(NSNotification *)aNotification
 {
     wxString st(wxMacConvertNewlines10To13(str));
     wxMacEditHelper helper(m_textView);
-    NSEvent* formerEvent = m_lastKeyDownEvent;
-    m_lastKeyDownEvent = nil;
+    wxWidgetCocoaKeyDownEvent* formerEvent = NULL;
+    PauseNativeKeyDownEvent(formerEvent);
     [m_textView insertText:wxCFStringRef( st , m_wxPeer->GetFont().GetEncoding() ).AsNSString()];
-    m_lastKeyDownEvent = formerEvent;
+    ResumeNativeKeyDownEvent(formerEvent);
```

Comment:
I.e. with my suggestion above these lines would become just
```
wxWidgetCocoaNativeKeyDownSuspend suspend(this);
[m_textView insertText:...];
```
and `m_lastKeyDownXXX` would be members of this `wxWidgetCocoaNativeKeyDownSuspend`.